### PR TITLE
ISSUE-2942 fix performance fallback

### LINKED ISSUE #2942: Performance fallback
(you don't have to strictly follow this form)

**Describe the situation**

The [performance doc](https://github.com/datafuselabs/databend/blob/main/website/databend/docs/overview/performance.md) can not be reproduced with latest code, about 0.3-0.5 X fallback.

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Fix performance fallback

## Changelog

- Performance Improvement

## Related Issues

fixes #2942 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by zhang2014** [comment]
```
Query:

mysql> SELECT sum(number) as sum FROM numbers_mt(1000000000) GROUP BY number % 3, number % 4, number % 5

Before:

Read 1000000000 rows, 8 GB in 2.631 sec., 380.06 million rows/sec., 3.04 GB/sec.

After:

Read 1000000000 rows, 8 GB in 0.677 sec., 1.48 billion rows/sec., 11.81 GB/sec.
```

File: query/src/datasources/database/system/processes_table.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ impl ProcessesTable {
             DataField::new("state", DataType::String, false),
             DataField::new("database", DataType::String, false),
             DataField::new("extra_info", DataType::String, true),
-            DataField::new("memory_usage", DataType::UInt64, true),
+            DataField::new("memory_usage", DataType::Int64, true),
```
Comment (sundy-li) [inline query/src/datasources/database/system/processes_table.rs:50]: why int64?

**Review by sundy-li — APPROVED** [review]

File: query/src/datasources/database/system/processes_table.rs (hunk 1 above)
Comment (zhang2014) [reply query/src/datasources/database/system/processes_table.rs:50]: Object may be created by thread 1 and destroy by thread 2

**Pull request merged**